app/models/announcement.rb

```diff
@@ -75,6 +78,11 @@ def reactions(account = nil)
     records
   end
 
+  def refresh_status_ids!
```

Comment:
The precedent for this kind of thing is `ProcessMentionsService`, that is, networking code isn't in the `Status` model but outside. Maybe best to keep it this way here too, you're only abstracting a single line of code behind this function, so the networking code could be directly in the `PublishScheduledAnnouncementWorker`. Does that make sense?